src/Type/Php/ReplaceFunctionsDynamicReturnTypeExtension.php

```diff
@@ -37,7 +39,29 @@ public function getTypeFromFunctionCall(
 		Scope $scope
 	): Type
 	{
-		$argumentPosition = $this->functions[$functionReflection->getName()];
+		$type = $this->getPreliminarilyResolvedTypeFromFunctionCall($functionReflection, $functionCall, $scope);
+
+		if ($this->functions[$functionReflection->getName()]['returnNull']) {
+			$type = $type instanceof UnionType
```

Comment:
What is the purpose of this branch?

Reply:
Add null to the result? What about `TypeCombinator::addNull()`?

Reply:
Add null to the result? - Yes
 What about TypeCombinator::addNull()? - fixed